tools/system_libs.py

```diff
@@ -496,9 +500,9 @@ def build_objects(self, build_dir):
       source_dir = utils.path_from_root()
       if batch_inputs:
         relative_source_dir = os.path.relpath(source_dir, build_dir)
-        cflags += [f'-ffile-prefix-map={relative_source_dir}/=']
-      cflags += [f'-ffile-prefix-map={source_dir}=/emsdk/emscripten',
-                 '-fdebug-compilation-dir=/emsdk/emscripten']
+        cflags += [f'-ffile-prefix-map={relative_source_dir}={DETERMINISITIC_PREFIX}']
+      cflags += [f'-ffile-prefix-map={source_dir}={DETERMINISITIC_PREFIX}',
+                 f'-fdebug-compilation-dir={DETERMINISITIC_PREFIX}']
```

Comment:
I wonder if these two chunks could be combined into `self.get_cflags()` rather than duplicated?   Can be part of the followup of course.

Reply:
Done in #23225.